include/zephyr/math/util.h

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file math/util.h
+ *
+ * @brief A common math library for use in both floating and fixed point.
+ */
+
+/*
+ * Copyright (c) 2022 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef INCLUDE_MATH_UTIL_H_
+#define INCLUDE_MATH_UTIL_H_
+
+#include <stdlib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef CONFIG_FPU
+#define PRIf	    "f"
+#define PRIf_ARG(n) (n)
+#else
+#define PRIf		    "d.%06llu"
+#define __FP_RAW_DECIMAL(n) ((int16_t)((n) & ((1 << FP_BITS) - 1)))
+#define __FP_DECIMAL(n)	    ((n < 0) ? (~__FP_RAW_DECIMAL(n) + 1) : __FP_RAW_DECIMAL(n))
+#define PRIf_ARG(n)                                                                                \
+	(FP_TO_INT(abs(n)) * (n < 0 ? -1 : 1)),                                                    \
+		((__FP_DECIMAL(n) * UINT64_C(1000000)) + ((1 << FP_BITS) / 2)) / (1 << FP_BITS)
+#endif
+
+#ifdef CONFIG_FPU
+typedef float fp_t;
+typedef float fp_inter_t;
+/* Conversion to/from fixed-point */
+#define INT_TO_FP(x)   ((float)(x))
+#define FP_TO_INT(x)   ((int32_t)(x))
+/* Float to fixed-point, only for compile-time constants and unit tests */
+#define FLOAT_TO_FP(x) ((float)(x))
```

Comment:
Float to FP and the result it still float.

Reply:
Right, the idea was that I was hoping to create a generic math lib that would work with floating point or fixed point. So the thinking was that if the CPU has floating point it would use that.

Would you consider the application just turning off FPU via `CONFIG_FPU=n`? I didn't consider the use case of an app having FPU enabled but wanting to use Fixed point.

Reply:
Look into my generic comment. This is not a good idea, as the CPU may use FPU but wish to use Fix Point for the communication. The decision if the real float or FP is used should be up to the module.